LUA : report aircraft configuration info

Pradeep-Carbonix opened the pull request:
reports aircraft configuration if "@ROMFS/AircraftConfiguration.xml" file exists.

robertlong13 commented on libraries/AP_HAL_ChibiOS/hwdef/CarbonixCommon/scripts/cx_built_in_test.lua:
```diff
@@ -59,8 +59,45 @@ assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 1), 'could not add '
 --]]
 local PREARM_BYPASS = bind_add_param('PREARM_DIS', 1, 0)
 
+-- fetches the aircraft configuration from the config file, if it exists
```
```suggestion
-- fetches the aircraft configuration from the config file and prints the model name and version
```

Pradeep-Carbonix replied:
Updated

robertlong13 commented on libraries/AP_HAL_ChibiOS/hwdef/CarbonixCommon/scripts/cx_built_in_test.lua:
```diff
@@ -59,8 +59,45 @@ assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 1), 'could not add '
+        if model and model_version then
+            cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "config (" .. model .. "_" .. model_version .. ".xml)")
+            return
+        end
```
Should have an else and print a message like you did above for if `line` is nil

Pradeep-Carbonix replied:
Avoided this print, because we are parsing an xml file.
But this is handled by this check at the top : 
```
        if not line then
            -- empty file or reached end of file
            cx_msg:send(cx_msg.MAV_SEVERITY.INFO, "missing model info in cfg")
            break
        end
```

robertlong13 reviewed (changes requested):
Just a couple small changes requested

robertlong13 commented on libraries/AP_HAL_ChibiOS/hwdef/CarbonixCommon/scripts/cx_built_in_test.lua:
```diff
@@ -59,8 +59,45 @@ assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 1), 'could not add '
+    local model = nil
+    local model_version = nil
+
+    -- Open file try and read header
```
Let's add a `logger:log_file_content(file_name)` here. That will get the config file into the logs.

Pradeep-Carbonix replied:
Thanks for this suggestion Bob, We now see the AircraftConfiguration.xml file embedded within logs:
![image](https://github.com/user-attachments/assets/76b2728c-0eb0-4c4d-8903-a01f92c327c8)

Note : As discussed, filenames in logs are limited to 16 characters. Hence we see - "AircraftConfigur" in UAV Log Viewer.

loki077 commented:
![image](https://github.com/user-attachments/assets/c8dde20b-48ed-46ef-8125-a8d4697cc5ac)
Test results with the modified code works.

loki077 committed:
LUA: Print Aircraft Configuration Version from XML
This commit adds a function to fetch the aircraft configuration from
the XML file and print the model name and version.

loki077 commented:
Test: Aircraftconfig.xml file in binary log. The name is limited to 16char

![image](https://github.com/user-attachments/assets/8d8fb81a-520a-4052-b26c-33a30526098e)
